src/main/java/tds/exam/results/mappers/OpportunityMapper.java

```diff
@@ -136,7 +136,7 @@ private static void mapExamItemsAndResponsesToOpportunity(final ExpandableExam e
             Item assessmentItem = assessmentItems.get(examItem.getItemKey());
 
             opportunityItem.setPosition(examItem.getPosition());
-            opportunityItem.setSegmentId(examSegments.get(examPage.getSegmentKey()).getSegmentId());
+            opportunityItem.setSegmentId(examPage.getSegmentKey());
```

Comment:
This is the important change.  We determined that the segment key is supposed to be used instead of the segment id.  S is session.testopportunitysegment [here|https://github.com/SmarterApp/TDS_TestDeliverySystemDataAccess/blob/master/tds-dll-mysql/src/main/java/tds/dll/mysql/ReportingDLL.java#L871]